src/sparse13/spsolve.cpp

```diff
@@ -269,98 +240,90 @@ ElementPtr  pPivot;
  *      Size of matrix. Made local to reduce indirection.
  *  Temp  (ComplexNumber)
  *      Temporary storage for entries in arrays.
- *
- *  >>> Obscure Macros
- *  IMAG_VECTORS
- *      Replaces itself with `, iRHS, iSolution' if the options spCOMPLEX and
- *      spSEPARATED_COMPLEX_VECTORS are set, otherwise it disappears
- *      without a trace.
  */
 
-static void
-SolveComplexMatrix( Matrix, RHS, Solution IMAG_VECTORS )
-
-MatrixPtr  Matrix;
-RealVector  RHS, Solution IMAG_VECTORS;
+static void SolveComplexMatrix(MatrixPtr Matrix, RealVector RHS, RealVector Solution, std::optional<RealVector> iRHS, std::optional<RealVector> iSolution)
```

Comment:
Apart from changing the function signatures from c to c++, the main data type change is to use `std::optional` instead of `IMAG_VECTORS` or `IMAG_RHS` which were defined in `spdefs.h`
```
#if spCOMPLEX AND spSEPARATED_COMPLEX_VECTORS
#define IMAG_VECTORS    , iRHS, iSolution
#define IMAG_RHS        , iRHS
#else
#define IMAG_VECTORS
#define IMAG_RHS
#endif
```